internal/clients/recommendations/client.go

```diff
@@ -98,24 +85,23 @@ func (rc *RecommendationsClient) request(method, url string, headers map[string]
 		req.Header.Add(k, v)
 	}
 
-	token := func() string {
-		rc.tokenMu.RLock()
-		defer rc.tokenMu.RUnlock()
-		return rc.accessToken
+	token, err := rc.tokener.NextToken(req.Context())
+	if err != nil {
+		return nil, err
 	}
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token()))
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
 
 	resp, err := rc.client.Do(req)
 	if err != nil {
 		return nil, err
 	}
 	if resp.StatusCode == http.StatusForbidden || resp.StatusCode == http.StatusUnauthorized {
-		err = rc.refreshToken()
+		token, err = rc.tokener.ForceRefresh(req.Context())
```

Comment:
I somehow do not like the pair `NextToken` and `ForceRefresh`. In the fist case, the call does not always return "next token", in fact it only does this once and then cache it. In the second case, I do not like that it does not contain `Token` in the method name.

Personal preference, but I would like something like `Token` and `RefreshToken` a bit more.